Sources/MapboxCoreNavigation/FeedbackEvent.swift

```
@@ -0,0 +1,34 @@
+import Foundation
+
+/**
+ Feedback event that can be created using `NavigationEventsManager.createFeedback()`.
+ Use `NavigationEventsManager.sendFeedback(_:type:description:)` to send it to the server.
+ Conforms to the `Codable` protocol, so the application can store the event persistently.
+ */
+public class FeedbackEvent: Codable {
+    let coreEvent: CoreFeedbackEvent
+    
+    init(eventDetails: NavigationEventDetails) {
+        let dictionary = try? eventDetails.asDictionary()
```

Should we really silently fail here?

---

Added an `assertionFailure`